packages/blade/src/components/Link/BaseLink/StyledBaseLink.web.tsx

```diff
@@ -1,14 +1,14 @@
 import styled from 'styled-components';
 import getStyledLinkStyles from './getStyledLinkStyles';
 import type { StyledBaseLinkProps } from './types';
-import { castWebType, getIn, makeBorderSize, makeMotionTime, useStyledProps } from '~utils';
+import { castWebType, getIn, makeBorderSize, makeMotionTime } from '~utils';
+import { useStyledProps } from '~components/Box/styled-props';
```

Comment:
useMemoizedStyles is our utility function that handles useMemo.

useStyledProps uses useMemoizedStyles and gives you the styled-props when any props object is passed to it.

Reply:
nahi samaj aaya 🙈

Reply:
So if I do -
```ts
const memoizedProps = useMemoizedStyles({ padding: '10px' });
```

memoizedProps will have `.padding` attribute as well (which is not supported on styled-props) because it doesn't exactly care what styles we're passing to it. `useMemoizedStyles` function in itself doesn't know what is supported on styled-props and what isn't because it is a common utility used by BaseBox as well as styledProps.

But, we want to make sure we don't expose `padding` or any other random prop to styledProps. So `useStyledProps` passes only the props that are supported on styled-props. Think of `useStyledProps` as a filter function which removes unnecessary props and limits the output to props that are supported on styledProps.